packages/shared/src/db.ts

```diff
@@ -86,23 +88,9 @@ declare const globalThis: {
 } & typeof global;
 
 if (process.env.NODE_ENV === "development") {
-  globalThis.prismaGlobal ??= new PrismaClient(); // regular instantiation
-  globalThis.kyselyPrismaGlobal ??= globalThis.prismaGlobal.$extends(
-    kyselyExtension({
-      kysely: (driver) =>
-        new Kysely<DB>({
-          dialect: {
-            // This is where the magic happens!
-            createDriver: () => driver,
-            // Don't forget to customize these to match your database!
-            createAdapter: () => new PostgresAdapter(),
-            createIntrospector: (db) => new PostgresIntrospector(db),
-            createQueryCompiler: () => new PostgresQueryCompiler(),
-          },
-        }),
-    })
-  );
-};
+  globalThis.prismaGlobal ??= PrismaClientSingleton.getInstance(true);
```

Comment:
I believe this will recreate the dev issues we saw previously. In this line, `prismaGlobal` will rely on the reference to the instance property on `PrismaClientSingleton` via `getInstance`. On hot reload, modules incl. the `shared` package will be reloaded such that the `PrismaClientSingleton.instance` will be undefined again, forcing the creation of a fresh client. Since on hot reload connections aren't closed gracefully, we will hit max_connections eventually. 

To fix the dev issues we have to create a separate client here (as previously done) or handle the `prismaGlobal` inside the PrismaClientSingleton to return the client bound to globalThis but not rely on the reference on `PrismaClientSingleton.instance`